src/linear_algebra/dimension.lean

```diff
@@ -74,22 +74,24 @@ begin
   exact cardinal.mk_emptyc (⊥ : submodule α β)
 end
 
-lemma dim_span {s : set β} (hs : linear_independent s) : dim α ↥(span s) = cardinal.mk s :=
-have (span s).subtype '' ((span s).subtype ⁻¹' s) = s :=
+set_option class.instance_max_depth 37
```

Comment:
So we get two `set_option class.instance_max_depth 37`. Where does this come from? Shouldn't the removal of `out_param` result in a easier type class search?

Reply:
The removal of out_param is only to enable modules over two rings, I believe. I don't know why we need to increase max depth. I believe it's from coercing a linear map to a function.